### ISSUE TITLE
Algorithm practice

### ISSUE THREAD

**Issue by scottbedard** [scottbedard/speedcube.site#12]
Now that #80 is approaching completion, it's time to start thinking more seriously about this feature. I want the platform to double as an algorithm training tool. This could even be useful as a training tool for real solving, depending on how natural the cube feels to a user based on their key bindings.

### Desired features

- [ ] Allow users to import their own algorithms, and group algorithms together in sets to practice. For example, someone might make an F2L set where they've picked from a bunch of algorithms they like, and can then train on those specific algs.
- [ ] Interactive scoreboard where users can see their average recognition and execution time for various algorithms.
- [ ] Ability to train in color-neutral mode
- [ ] Ability to train with randomized setup moves. This will let us practice algorithms from different angles, cutting cube-rotations and unnecessary moves from our solving.

### Nice to haves

- Badges on the user's profile showing that they've mastered some set of algorithms. This can help us gamify things a bit, and reward players for achieving major milestones like learning all the PLL algs, etc...

**Comment by scottbedard** [comment]
Blocked by #80
